lib/logstash/filters/split.rb

```diff
@@ -66,6 +66,8 @@ def register
   public
   def filter(event)
     
+    # Convert field to a string based on any format values    
+    @field = event.sprintf(@field)
```

Comment:
The result of `event.sprintf(@field)` is, naturally, event dependent. Therefore you can't save it back to `@field`. However, you should be able to just replace `@field` in the next statement with `event.sprintf(@field)`.